packages/esm-appointments-app/src/calendar/monthly/monthly-workload-view.component.tsx

```diff
@@ -26,56 +26,68 @@ const MonthlyWorkloadView: React.FC<MonthlyWorkloadViewProps> = ({ dateTime, eve
   const onClick = (serviceUuid) => {
     navigate({ to: `${spaHomePage}/appointments/${dayjs(dateTime).format('YYYY-MM-DD')}/${serviceUuid}` });
   };
-
-  return (
-    <div
-      onClick={() => {
-        onClick('');
-      }}
-      className={classNames(
-        styles[isSameMonth(dateTime, dayjs(selectedDate)) ? 'monthly-cell' : 'monthly-cell-disabled'],
-        {
-          [styles.greyBackground]: currentData?.services,
-          [styles.smallDesktop]: layout === 'small-desktop',
-          [styles.largeDesktop]: layout !== 'small-desktop',
-        },
-      )}>
-      {isSameMonth(dateTime, dayjs(selectedDate)) && (
-        <p>
-          <b className={styles.calendarDate}>{dateTime.format('D')}</b>
-          {currentData?.services && (
-            <div className={styles.currentData}>
-              {currentData.services.map(({ serviceName, serviceUuid, count }, i) => (
+  if (!currentData) {
+    return (
+      <div
+        className={classNames(
+          styles[isSameMonth(dateTime, dayjs(selectedDate)) ? 'monthly-cell' : 'monthly-cell-disabled'],
+          {
+            [styles.greyBackground]: currentData?.services,
+            [styles.smallDesktopEmpty]: layout === 'small-desktop',
+            [styles.largeDesktopEmpty]: layout !== 'small-desktop',
+          },
+        )}>
+        {isSameMonth(dateTime, dayjs(selectedDate)) && <b className={styles.calendarDate}>{dateTime.format('D')}</b>}
+      </div>
+    );
+  } else {
```

Comment:
Just a single check for currentData while calling isSameMonth() should be fine, no need for if-else clause.
Rest looks great.